source/base/data_out_base.cc

```diff
@@ -2700,6 +2705,22 @@ namespace DataOutBase
 
     for (const auto &patch : patches)
       {
+        if (patch.reference_cell_type != ReferenceCell::get_hypercube(dim))
+          {
+            connectivity.resize(patch.data.n_cols());
+
+            for (unsigned int i = 0; i < patch.data.n_cols(); ++i)
+              connectivity[i] = i;
```

Comment:
If we choose to have a different ordering, we need to adjust the permutation here.